core/chains/evm/client/chain_client.go

```diff
@@ -105,96 +103,151 @@ func (c *chainClient) BlockByHash(ctx context.Context, hash common.Hash) (b *typ
 
 func (c *chainClient) CodeAt(ctx context.Context, account common.Address, blockNumber *big.Int) ([]byte, error) {
-	return c.multiNode.CodeAt(ctx, account, blockNumber)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.CodeAt(ctx, account, blockNumber)
 }
 
 func (c *chainClient) ConfiguredChainID() *big.Int {
-	return c.multiNode.ConfiguredChainID()
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil
+	}
+	return rpc.chainID
 }
 
 func (c *chainClient) Dial(ctx context.Context) error {
-	return c.multiNode.Dial(ctx)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return err
+	}
+	return rpc.Dial(ctx)
 }
 
 func (c *chainClient) EstimateGas(ctx context.Context, call ethereum.CallMsg) (uint64, error) {
-	return c.multiNode.EstimateGas(ctx, call)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return 0, err
+	}
+	return rpc.EstimateGas(ctx, call)
 }
 func (c *chainClient) FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error) {
-	return c.multiNode.FilterEvents(ctx, q)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.FilterEvents(ctx, q)
 }
 
 func (c *chainClient) HeaderByHash(ctx context.Context, h common.Hash) (head *types.Header, err error) {
-	rpc, err := c.multiNode.SelectNodeRPC()
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
 		return head, err
 	}
 	return rpc.HeaderByHash(ctx, h)
 }
 
 func (c *chainClient) HeaderByNumber(ctx context.Context, n *big.Int) (head *types.Header, err error) {
-	rpc, err := c.multiNode.SelectNodeRPC()
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
 		return head, err
 	}
 	return rpc.HeaderByNumber(ctx, n)
 }
 
 func (c *chainClient) HeadByHash(ctx context.Context, h common.Hash) (*evmtypes.Head, error) {
-	return c.multiNode.BlockByHash(ctx, h)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.BlockByHash(ctx, h)
 }
 
 func (c *chainClient) HeadByNumber(ctx context.Context, n *big.Int) (*evmtypes.Head, error) {
-	return c.multiNode.BlockByNumber(ctx, n)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.BlockByNumber(ctx, n)
 }
 
 func (c *chainClient) IsL2() bool {
-	return c.multiNode.IsL2()
+	return c.multiNode.ChainType().IsL2()
 }
 
 func (c *chainClient) LINKBalance(ctx context.Context, address common.Address, linkAddress common.Address) (*commonassets.Link, error) {
-	return c.multiNode.LINKBalance(ctx, address, linkAddress)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.LINKBalance(ctx, address, linkAddress)
 }
 
 func (c *chainClient) LatestBlockHeight(ctx context.Context) (*big.Int, error) {
-	return c.multiNode.LatestBlockHeight(ctx)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return nil, err
+	}
+	return rpc.LatestBlockHeight(ctx)
 }
 
 func (c *chainClient) NodeStates() map[string]string {
```

Comment:
It might be better to change the signature to this method to also use NodeState. Unless it significantly increases scope of the PR